firewall/privacy_mapper_test.go

```diff
@@ -718,6 +719,196 @@ func TestHideBool(t *testing.T) {
+				`["d23da57575cdcb878ac191e1e0c8a5c4f061b11cfdc7a8ec5c9d495270de66fdbf",` +
+				`"0e092708c9e737115ff14a85b65466561280d77c1b8cd666bc655536ad81ccca85",` +
+				`"DEAD2708c9e737115ff14a85b65466561280d77c1b8cd666bc655536ad81ccca85",` +
+				`"0e092708c9e737115ff14a85b65466561280d77c1b8cd666bc655536ad81ccca85",` +
+				`"586b59212da4623c40dcc68c4573da1719e5893630790c9f2db8940fff3efd8cd4"]}`),
+			knownMap: map[string]string{
+				"586b59212da4623c40dcc68c4573da1719e5893630790c9f2db8940fff3efd8cd4": "123456789012345678901234567890123456789012345678901234567890123456",
+			},
+			expectedNewPairs: 3,
+		},
+		{
+			// We don't substitute unknown items.
+			name: "all invalid pubkeys",
+			config: []byte(`{"version":1,"list":` +
+				`["d23da57575cdcb878ac191e1e0c8a5c4f061b11",` +
+				`"586b59212da4623c40dcc68c4573da1719e5893630790c9f2db8940fff3efd8cd4dead",` +
+				`"x86b59212da4623c40dcc68c4573da1719e5893630790c9f2db8940fff3efd8cd4"]}`),
+			expectedNewPairs: 0,
+		},
+		{
+			// We only substitute channel ids that have a sane
+			// format.
+			name: "channel ids",
+			config: []byte(`{"version":1,"list":` +
+				`["1",` +
+				`"12345",` +
+				`"1234567890123",` +
+				`"1234567890123456789",` +
+				`"123456789012345678901"]}`),
+			expectedNewPairs: 2,
+		},
+		{
+			// We obfuscate channel points, the character length may
+			// vary due to the output index.
+			name: "channel points",
+			config: []byte(`{"version":1,"list":` +
+				`["0e092708c9e737115ff14a85b65466561280d77c1b8cd666bc655536ad81ccca:1",` +
+				`"e092708c9e737115ff14a85b65466561280d77c1b8cd666bc655536ad81ccca:1",` +
+				`"0e092708c9e737115ff14a85b65466561280d77c1b8cd666bc655536ad81ccca3:1",` +
+				`"0e092708c9e737115ff14a85b65466561280d77c1b8cd666bc655536ad81ccca:3000"]}`),
+			expectedNewPairs: 2,
+			notExpectSameLen: true,
+		},
+		{
+			// We only act on items that are in lists of strings.
+			name: "single pubkey with another field",
+			config: []byte(`{"version":1,"list":` +
+				`["586b59212da4623c40dcc68c4573da1719e5893630790c9f2db8940fff3efd8cd4"],` +
+				`"another":"0e092708c9e737115ff14a85b65466561280d77c1b8cd666bc655536ad81ccca85"}`),
+			expectedNewPairs: 1,
+		},
+		{
+			// We don't obfuscate any numbers even though they may
+			// be channel ids. This is to be able to set numerical
+			// values in the range of channel ids.
+			name:             "number",
+			config:           []byte(`{"version":1,"number":12345678901234567890}`),
+			expectedNewPairs: 0,
+		},
+		{
+			// We don't allow to mix strings with other types, which
+			// may be a configuration mistake.
+			name: "list of invalid types",
+			config: []byte(`{"version":1,"channels":` +
+				`[12345,` +
+				`"e092708c9e737115ff14a85ab65466561280d77c1b8cd666bc655536ad81ccca:1"]}`),
+			expectErr:        true,
+			expectedNewPairs: 0,
+		},
+		{
+			// A list of numbers is not obfuscated. Those can be
+			// useful to submit histograms for example.
+			name: "channel ids",
```

Comment:
we would want to obfuscate channel IDs though. Even though they are numbers. This makes me think we should allow the user to specify specify somehow that a list/item in a config struct should be obfuscated. We need to think about how to do this though... We can do this in future though :) we just need to do it before ever asking for channel IDs in the config

Reply:
yes, agree that we need to think more about these cases and that such a signal could make sense, for now the signal for obfuscation is to have a list of strings, which includes the channel id case as well, just the numbers converted to strings